server/methods/getGuestAccount.js

```diff
@@ -0,0 +1,27 @@
+Meteor.methods({
+	getGuestAccount: () => {
+		let guestId = 1;
+		let lastGuest = RocketChat.models.Users.findOne({guestId: {$gt: 0}}, {sort: {DateTime: -1, limit: 1}});
+		if (lastGuest) {
+			guestId = lastGuest.guestId + 1;
+		}
+		let username = 'anonymous-' + guestId;
+		while (RocketChat.models.Users.findOne({ username: username })) {
```

Comment:
I am not a fan of this. I feel like the logic should something like "count how many `anonymous-` users there are and add one, then check if that one exists and add up from there". That way we aren't starting from `anonymous-1` each and every time, but starts from the amount of users + 1.